pkg/alertmanager/multitenant.go

```diff
@@ -479,10 +479,46 @@ func (am *MultitenantAlertmanager) ServeHTTP(w http.ResponseWriter, req *http.Re
 	userAM, ok := am.alertmanagers[userID]
 	am.alertmanagersMtx.Unlock()
 
-	if !ok || !userAM.IsActive() {
+	if !ok && am.fallbackConfig != "" {
+		// Give the user the fallback configuration and start its alertmanager
+		userAmConfig, err := amconfig.Load(am.fallbackConfig)
+		if err != nil {
+			level.Error(am.logger).Log("msg", "failed to load the fallback configuration", "user", userID, "err", err)
+			http.Error(w, "Failed to initialize the Alertmanager", http.StatusInternalServerError)
+		}
+
+		transformedConfig, err := am.transformConfig(userID, userAmConfig)
+		if err != nil {
+			level.Error(am.logger).Log("msg", "failed to transform the configuration", "user", userID, "err", err)
+			http.Error(w, "Failed to initialize the Alertmanager", http.StatusInternalServerError)
+		}
+
+		newAM, err := am.newAlertmanager(userID, transformedConfig)
+		if err != nil {
+			level.Error(am.logger).Log("msg", "failed to start the Alertmanager", "user", userID, "err", err)
+			http.Error(w, "Failed to initialize the Alertmanager", http.StatusInternalServerError)
+		}
+
+		am.alertmanagersMtx.Lock()
+		am.alertmanagers[userID] = newAM
```

Comment:
There's a possible race condition here. Before overriding the alertmanager config for the user, you should check that it hasn't been already set in the meanwhile (between the previous lock and this lock).